src/upf_network.py

```diff
@@ -49,12 +47,82 @@ def get_ip_address(self) -> str:
         return ""
 
+    def addresses_are_set(self) -> bool:
+        """Check if the given network interface is already configured."""
+        interfaces = self.network_db.interfaces  # type: ignore[reportAttributeAccessIssue]
+        try:
+            iface_record = interfaces[self.name]
+        except KeyError:
+            logger.warning("Interface %s not found in the network database", self.name)
+            return False
+        ip_addresses = iface_record.ipaddr
+        for ip in ip_addresses:
+            if ip.family == AF_INET:
+                logger.info("Found IP: %s/%s", ip.address, ip.prefixlen)
+                if f"{ip.address}/{ip.prefixlen}" != self.ip_address:
+                    return False
+        if not self.get_ip_address():
+            return False
+        return True
+
+    def set_ip_address(self) -> None:
+        """Clean all unrequired IPs and set the IP address for the given network interface."""
+        interfaces = self.network_db.interfaces  # type: ignore[reportAttributeAccessIssue]
+        try:
+            iface_record = interfaces[self.name]
+        except KeyError:
+            logger.warning("Interface %s not found in the network database", self.name)
+            return
+        ip_addresses = iface_record.ipaddr
+        # remove all unrequired IPs
+        for ip in ip_addresses:
+            if ip.family == AF_INET:
+                if ip.address != self.ip_address:
+                    logger.info("Removing IP %s/%s from interface %s",
+                                ip.address, ip.prefixlen, self.name)
+                    iface_record.del_ip(f"{ip.address}/{ip.prefixlen}").commit()
+        # add requested IP if not already there
+        if not self.get_ip_address():
+            logger.info("Adding IP %s to interface %s", self.ip_address, self.name)
+            iface_record.add_ip(self.ip_address).commit()
+
+    def unset_ip_address(self) -> None:
+        """Remove the configured IP address from the given network interface."""
+        interfaces = self.network_db.interfaces  # type: ignore[reportAttributeAccessIssue]
+        try:
+            iface_record = interfaces[self.name]
+        except KeyError:
+            logger.warning("Interface %s not found in the network database", self.name)
+            return
+        logger.info("Removing IP %s from interface %s", self.ip_address, self.name)
+        iface_record.del_ip(self.ip_address).commit()
+
+    def mtu_size_is_set(self) -> bool:
+        """Check if MTU size of the given network interface is already configured ."""
+        interfaces = self.network_db.interfaces  # type: ignore[reportAttributeAccessIssue]
+        try:
+            iface_record = interfaces[self.name]
+            return iface_record.get("mtu") == self.mtu_size
+        except KeyError:
+            logger.warning("Interface %s not found in the network database", self.name)
+            return False
```

Comment:
Instead of having a bunch of try/excepts for key errors, I recommend using `interfaces.get(self.name, None)`. Here we have a mix of the two approaches one line after the other:

```python
interfaces[self.name]
iface_record.get("mtu") 
```
This applies at a couple of places